packages/@aws-cdk/aws-codepipeline/lib/private/stage.ts

```diff
@@ -174,3 +152,13 @@ function sanitizeArtifactName(artifactName: string): string {
   // but not in Artifact names
   return artifactName.replace(/[@.]/g, '');
 }
+
+function* duplicateStrings(xs: string[]): IterableIterator<string> {
```

Comment:
Not sure I see the point of this generator here...? Aren't we just creating a Set from the passed array...?

Reply:
No.

I would hope the function name is a hint 😛